lib/livebook/session.ex

```diff
@@ -1853,6 +1854,16 @@ defmodule Livebook.Session do
     {:noreply, handle_operation(state, operation)}
   end
 
+  def handle_info({:hub_changed, id}, %{data: %{notebook: %{hub_id: id}}} = state) do
+    if Livebook.Hubs.hub_exists?(id) do
+      {:noreply, state}
+    else
+      # Since the hub got deleted, all notebooks that belongs to the deleted hub and has open sessions must be closed.
```

Comment:
```suggestion
      # Since the hub got deleted, we close all sessions using that hub.
      # This way we clean up all secrets and other in-memory state that
      # is related to the hub
```